monkey/monkey_island.py

```diff
@@ -16,10 +18,15 @@
     # This is here in order to catch EVERYTHING, some functions are being called on
     # imports, so the log init needs to be first.
     try:
-        server_config_path = os.path.expanduser(island_args.server_config)
+        if island_args.server_config:
+            server_config_path = os.path.expanduser(island_args.server_config)
+        else:
+            server_config_path = server_config_generator.create_default_server_config_file()
 
         config = config_loader.load_server_config_from_file(server_config_path)
 
+        create_data_dir(config["data_dir"], True)
```

Comment:
This line only makes sense if `island_args.server_config`, because otherwise you create the dir in `server_config_generator.create_default_server_config_file()`

Reply:
Fixed in #1184